src/schematools/events/export.py

```diff
@@ -0,0 +1,143 @@
+            if (relation := field.relation) is not None:
+                multi = False
+                properties = field["properties"]
+                relation_ds = relation.split(":")[0]
+        except KeyError:
+            continue
+
+        if multi is not None:
+            complex_fields[to_snake_case(field.name)] = {
+                "multi": multi,
+                "relation_ds": relation_ds,
+                "identifier_names": dataset_table.identifier,
+                "sub_field_names": [to_snake_case(sf) for sf in properties.keys()],
+            }
+    return complex_fields
+
+
+def collect_nm_embed_rows(
+    dataset_id: str,
+    table_id: str,
+    datasets_lookup: dict[str, DatasetSchema],
+    tables: dict[str, dict[str, Table]],
+    complex_fields_info: dict[str, dict],
+    connection: Connection,
+) -> dict[str:Any]:
+    """Fetch row info as list of embeddable objects."""
+    nm_embeds = defaultdict(lambda: defaultdict(list))
+    for field_name, field_info in complex_fields_info.items():
+        if field_info["multi"]:
+            through_table = tables[dataset_id][f"{table_id}_{field_name}"]
+            for row in connection.execute(through_table.select()):
+                row_dict = dict(row)
+                id_value = ".".join(
+                    str(row_dict[f"{table_id}_{idn}"]) for idn in field_info["identifier_names"]
+                )
+
+                stripped_row = {}
+                for sfn in field_info["sub_field_names"]:
+                    stripped_row[sfn] = row_dict[f"{field_name}_{sfn}"]
+                nm_embeds[table_id][id_value].append(stripped_row)
+    return nm_embeds
+
+
+def fetch_nm_embeds(
+    row: dict[str, Any],
+    table_id: str,
+    nm_embed_rows: dict[str, Any],
+    complex_fields_info: dict[str, dict],
+) -> dict[str, list[dict[str, Any]]]:
+    """Fetch row info as lists of embeddables."""
+    nm_embeds = defaultdict(list)
+    for field_name, field_info in complex_fields_info.items():
+        if field_info["multi"]:
+            id_value = ".".join(str(row[idn]) for idn in field_info["identifier_names"])
+            row_dicts = nm_embed_rows[table_id].get(id_value)
+            nm_embeds[field_name] = row_dicts
+
+    return nm_embeds
+
+
+def fetch_1n_embeds(
+    row: dict[str, Any], complex_fields_info: dict[str, dict]
+) -> dict[str, dict[str, Any]]:
+    """Fetch row info as embeddable object(s)."""
+    embeddable_objs = {}
+    for field_name, field_info in complex_fields_info.items():
+        if not field_info["multi"]:
+            embed_obj = {}
+            for sub_field_name in field_info["sub_field_names"]:
+                embed_obj[sub_field_name] = row[f"{field_name}_{sub_field_name}"]
+            embeddable_objs[field_name] = embed_obj
+    return embeddable_objs
+
+
+def export_events(
+    datasets, dataset_id: str, table_id: str, connection: Connection
+) -> Generator[str, None, None]:
```

Comment:
If you are only specifying `YieldType` as you do here, you should probably use `-> Iterator[str]`. It is more succint.

Reply:
👍🏼